CMakeLists.txt

```diff
@@ -629,6 +629,19 @@ if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC" OR "${CMAKE_CXX_SIMULATE_ID}" STREQU
   set(SWIFT_COMPILER_IS_MSVC_LIKE TRUE)
 endif()
 
+if(NOT SWIFT_COMPILER_IS_MSVC_LIKE)
+  # Avoid CMake's default -O3 for Release builds.
+  set(CMAKE_CXX_FLAGS_RELEASE "-O2 -DNDEBUG")
```

Comment:
Please don't set `-DNDEBUG`, that overrides the behaviour of `LLVM_ENABLE_ASSERTIONS`.  I usually build with assertions intentionally.

Reply:
It is the default in CMAKE_CXX_FLAGS_RELEASE, though. I though it would be better to keep it that way, because we're explicitly checking LLVM_ENABLE_ASSERTIONS and passing `-UNDEBUG` or `-DNDEBUG` after we include CMAKE_CXX_FLAGS_${CMAKE_BUILD_TYPE}.